src/model/HeatPumpPlantLoopEIRCooling.cpp

```diff
@@ -0,0 +1,491 @@
+                               electricInputtoOutputRatioModifierFunctionofTemperatureCurve.handle());
+      return result;
+    }
+
+    bool HeatPumpPlantLoopEIRCooling_Impl::setElectricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve(
+      const CurveQuadratic& electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve) {
+      bool result = setPointer(OS_HeatPump_PlantLoop_EIR_CoolingFields::ElectricInputtoOutputRatioModifierFunctionofPartLoadRatioCurveName,
+                               electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve.handle());
+      return result;
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::autosizedReferenceLoadSideFlowRate() const {
+      return getAutosizedValue("Design Size Load Side Volume Flow Rate", "m3/s");
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::autosizedReferenceSourceSideFlowRate() const {
+      return getAutosizedValue("Design Size Source Side Volume Flow Rate", "m3/s");
+    }
+
+    boost::optional<double> HeatPumpPlantLoopEIRCooling_Impl::autosizedReferenceCapacity() const {
+      return getAutosizedValue("Design Size Nominal Capacity", "W");
+    }
+
+    void HeatPumpPlantLoopEIRCooling_Impl::autosize() {
+      autosizeReferenceLoadSideFlowRate();
+      autosizeReferenceSourceSideFlowRate();
+      autosizeReferenceCapacity();
+    }
+
+    void HeatPumpPlantLoopEIRCooling_Impl::applySizingValues() {
+      boost::optional<double> val;
+      val = autosizedReferenceLoadSideFlowRate();
+      if (val) {
+        setReferenceLoadSideFlowRate(val.get());
+      }
+
+      val = autosizedReferenceSourceSideFlowRate();
+      if (val) {
+        setReferenceSourceSideFlowRate(val.get());
+      }
+
+      val = autosizedReferenceCapacity();
+      if (val) {
+        setReferenceCapacity(val.get());
+      }
+    }
+
+  }  // namespace detail
+
+  HeatPumpPlantLoopEIRCooling::HeatPumpPlantLoopEIRCooling(const Model& model, const CurveBiquadratic& capacityModifierFunctionofTemperatureCurve,
+                                                           const CurveBiquadratic& electricInputtoOutputRatioModifierFunctionofTemperatureCurve,
+                                                           const CurveQuadratic& electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve)
+    : WaterToWaterComponent(HeatPumpPlantLoopEIRCooling::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::HeatPumpPlantLoopEIRCooling_Impl>());
+
+    autosizeReferenceLoadSideFlowRate();
+    autosizeReferenceSourceSideFlowRate();
+    autosizeReferenceCapacity();
+
+    bool ok = setCapacityModifierFunctionofTemperatureCurve(capacityModifierFunctionofTemperatureCurve);
+    OS_ASSERT(ok);
+
+    ok = setElectricInputtoOutputRatioModifierFunctionofTemperatureCurve(electricInputtoOutputRatioModifierFunctionofTemperatureCurve);
+    OS_ASSERT(ok);
```

Comment:
So here I'd take Curve& instead of Biquad/Quad, and appropriately handle the case where it fails via remove() + log and throw